HWs/HW2/StackCalculator/StackCalculator/Stack/StackArray.cs

```diff
@@ -0,0 +1,48 @@
+namespace StackCalculator;
+
+/// <summary>
+/// Stack realization by using array.
+/// </summary>
+public class StackArray : IStack
+{
+    public StackArray()
+    {
+        this.stack = new List<double>();
+    }
+
+    private List<double> stack;
+
+    private int length = 0;
+
+    /// <inheritdoc/>
+    public void Push(double value)
+    {
+        if (length == stack.Count)
+        {
+            stack.Add(value);
+        }
+        else
+        {
+            stack[length] = value;
+        }
+        ++length;
```

Comment:
```suggestion
        stack.Add(value);
        ++length;
```
Вставка элемента в список уже оптимизирована ([proof](https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/mscorlib/system/collections/generic/list.cs#L220)).
Так что контроль размера и вставка только в случае переполнения избыточна.